File: auidoRecorderStep1/app/src/main/java/com/example/auidorecorderstep1/MainActivity.java
package com.example.auidorecorderstep1;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.media.AudioFormat;
import android.os.Bundle;
import android.os.StrictMode;
import android.view.View;
import android.widget.Button;

import com.github.squti.androidwaverecorder.WaveRecorder;

public class MainActivity extends AppCompatActivity {


    Button btnStart;
    Button btnStop;
    Button btnPause;

    WaveRecorder waveRecorder;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);


        InitializeWidgets();
        InitializeEvents();


        String filePath = getExternalCacheDir().getAbsolutePath() + "/audioFile.wav";

        waveRecorder = new WaveRecorder(filePath);

        waveRecorder.getWaveConfig().setSampleRate(44100);
        waveRecorder.getWaveConfig().setChannels(AudioFormat.CHANNEL_IN_STEREO);
        waveRecorder.getWaveConfig().setAudioEncoding(AudioFormat.ENCODING_PCM_8BIT);
        waveRecorder.setNoiseSuppressorActive(true);
        System.out.println("file path: " + filePath);



    }

    public void InitializeWidgets(){
        btnStart = findViewById(R.id.btnStart);
        btnStop = findViewById(R.id.btnStop);
        btnPause = findViewById(R.id.btnPause);
    }


    public void InitializeEvents(){

        btnStart.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                waveRecorder.startRecording();
                System.out.println("start");

            }
        });

        btnPause.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                System.out.println("pause");

            }
        });

        btnStop.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                waveRecorder.startRecording();
                System.out.println("stop");

            }
        });

    }



}

File: AudioRecorderStep1inStep2/app/src/main/java/com/example/audiorecorderstep1instep2/MainActivity.kt
package com.example.audiorecorderstep1instep2

import android.content.pm.PackageManager
import android.media.AudioFormat
import android.nfc.Tag
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.os.Environment
import android.util.Log
import android.widget.Toast
import androidx.core.app.ActivityCompat
import androidx.core.content.ContextCompat
import com.github.squti.androidwaverecorder.WaveRecorder
import kotlinx.android.synthetic.main.activity_main.*
import java.io.File
import java.util.*
import java.util.jar.Manifest

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var dizi = arrayOf(android.Manifest.permission.RECORD_AUDIO,android.Manifest.permission.WRITE_EXTERNAL_STORAGE,android.Manifest.permission.READ_EXTERNAL_STORAGE)

        if (ContextCompat.checkSelfPermission(this,
                        android.Manifest.permission.RECORD_AUDIO)
                != PackageManager.PERMISSION_GRANTED) {
            ActivityCompat.requestPermissions(this,
                    dizi,
                    1234);
        }

        var guid = UUID.randomUUID().toString();
        var file = File(Environment.getExternalStorageDirectory(), "/Music/"+guid+".mp3")
        //val filePath:String = externalCacheDir?.absolutePath + "/Music/audioFile.wav"
        var filePath:String = file.absolutePath



        var waveRecorder = WaveRecorder(filePath )
        waveRecorder.waveConfig.sampleRate = 16000
        waveRecorder.waveConfig.channels = AudioFormat.CHANNEL_IN_MONO
        waveRecorder.waveConfig.audioEncoding = AudioFormat.ENCODING_PCM_16BIT
        waveRecorder.noiseSuppressorActive = true






        btnStart.setOnClickListener {

            var guid = UUID.randomUUID().toString();
            var file = File(Environment.getExternalStorageDirectory(), "/Music/"+guid+".mp3")
            //val filePath:String = externalCacheDir?.absolutePath + "/Music/audioFile.wav"
            var filePath:String = file.absolutePath



            //val filePath:String = externalCacheDir?.absolutePath + "/Music/audioFile.wav"
            waveRecorder = WaveRecorder(filePath )
            waveRecorder.waveConfig.sampleRate = 16000
            waveRecorder.waveConfig.channels = AudioFormat.CHANNEL_IN_MONO
            waveRecorder.waveConfig.audioEncoding = AudioFormat.ENCODING_PCM_16BIT
            waveRecorder.noiseSuppressorActive = true


            Toast.makeText(this,"basladi",Toast.LENGTH_SHORT).show()
            waveRecorder.startRecording()


        }

        btnPause.setOnClickListener {

            waveRecorder.pauseRecording()
            Toast.makeText(this,"durdu",Toast.LENGTH_SHORT).show()

        }

        btnStop.setOnClickListener {
            waveRecorder.stopRecording()
            Toast.makeText(this,"bitti",Toast.LENGTH_SHORT).show()

        }



    }


    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults)


        if(requestCode == 1234){
            if (grantResults.size > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {


                Toast.makeText(this,"izin var",Toast.LENGTH_SHORT).show()


            }

        }

    }

}